### PR TITLE
Holder PR for migration merge

### PR DESCRIPTION
**Pull request by mishra14**
Merging migration into dev today. 

This PR is to allow people to take a look before we merge. If needed please raise a concern.

Changes - 
1. Migrated most of the projects to .Net Core. The only projects left out are WPF related.
2. Merged NuGet.Core.Sln and NuGet.Client.Sln into NuGet.Sln.
3. Created build.proj to perform all of our build and test related tasks.
4. Changed our ps1 scripts to invoke build.proj.
5. Restructured props and targets to reflect the new project structures.

//cc: @alpaix @rohit21agrawal @emgarten @nkolev92 @jainaashish @zhili1208 @joelverhagen @rrelyea

### CODE REVIEW

File: .nuget/packages.config [hunk 1]
```diff
@@ -8,10 +8,13 @@
   <package id="Microsoft.VisualStudio.Setup.Configuration.Interop" version="1.1.142-preview4-29265" targetFramework="net35" />
   <package id="Microsoft.VisualStudio.Shell.Interop.15.0.DesignTime" version="15.0.0" targetFramework="net45" />
   <package id="Microsoft.VSSDK.BuildTools" version="14.3.25420" targetFramework="net45" />
-  <package id="Microsoft.VSSDK.BuildTools" version="15.0.26109-RC3" targetFramework="net46" />
+  <package id="Microsoft.VSSDK.BuildTools" version="15.0.26201" targetFramework="net46" />
```
Comment (nkolev92) [inline .nuget/packages.config:11]: This half-way fixes.
Need to make the change in the loc-sign pipeline:
https://github.com/NuGet/Home/issues/4727

File: .nuget/packages.config [hunk 2]
```diff
@@ -8,10 +8,13 @@
...
+  <package id="Microsoft.Web.Xdt" version="2.1.1" targetFramework="net40" />
   <package id="NuGet.Core" version="2.14.0-rtm-832" targetFramework="net45"/>
   <package id="NuGet.MsBuild.Integration" version="3.1.0-beta-001" targetFramework="net45" />
   <package id="VSLangProj150" version="1.0.0" targetFramework="net46" />
-  <package id="xunit.runner.console" version="2.1.0" targetFramework="net45" />
-  <package id="xunit.runner.msbuild" version="2.1.0" targetFramework="net45" />
-</packages>
+  <package id="xunit.runner.console" version="2.2.0" targetFramework="net45" />
+  <package id="xunit.runner.msbuild" version="2.2.0" targetFramework="net45" />
+</packages>
```
Comment (nkolev92) [inline .nuget/packages.config]: nit: 
new line

Comment (mishra14) [inline .nuget/packages.config]: up for grabs!

**Comment by mishra14** [comment]
Fixes: https://github.com/NuGet/Home/issues/3616

**Comment by nkolev92** [comment]
The VSIX that the migrate branch currently produces is corrupt.

**Review by nkolev92 — CHANGES_REQUESTED** [review]
Make sure the VSIX is packed correctly.

**Review by nkolev92 — APPROVED** [review]

**Review by emgarten — APPROVED** [review]

**Pull request merged**